doc/whats_new/v0.20.rst

```diff
@@ -25,6 +25,13 @@ random sampling procedures.
 Changelog
 ---------
 
+:mod:`sklearn.compose`
+......................
+
+- |Fix| Fixed an issue in :class:`compose.ColumnTransformer` which raises
+  unexpected error when columns is pandas Index.
```

Comment:
also pandas Series (although Index is more logical to pass here, so maybe it is fine to only mention that)